opendcs-rest-api/src/main/java/org/opendcs/odcsapi/sec/basicauth/BasicAuthResource.java

```diff
@@ -0,0 +1,222 @@
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.opendcs.odcsapi.sec.basicauth;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Base64;
+import java.util.Set;
+import javax.annotation.security.RolesAllowed;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.ServerErrorException;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.opendcs.odcsapi.dao.DbException;
+import org.opendcs.odcsapi.errorhandling.WebAppException;
+import org.opendcs.odcsapi.hydrojson.DbInterface;
+import org.opendcs.odcsapi.sec.AuthorizationCheck;
+import org.opendcs.odcsapi.sec.OpenDcsApiRoles;
+import org.opendcs.odcsapi.sec.OpenDcsPrincipal;
+import org.opendcs.odcsapi.util.ApiHttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/")
+public class BasicAuthResource
+{
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(BasicAuthResource.class);
+	private static final String MODULE = "BasicAuthResource";
+
+	@Context private HttpServletRequest request;
+	@Context private HttpHeaders httpHeaders;
+
+	@POST
+	@Path("credentials")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	@RolesAllowed({AuthorizationCheck.ODCS_API_GUEST})
+	public Response postCredentials(Credentials credentials) throws WebAppException
+	{
+		if(!DbInterface.isOpenTsdb)
+		{
+			throw new ServerErrorException("Basic Auth is not supported", Response.Status.NOT_IMPLEMENTED);
+		}
+
+		//If credentials are null, Authorization header will be checked.
+		if(credentials != null)
+		{
+			verifyCredentials(credentials);
+		}
+
+		String authorizationHeader = httpHeaders.getHeaderString(HttpHeaders.AUTHORIZATION);
+		credentials = getCredentials(credentials, authorizationHeader);
+		validateDbCredentials(credentials);
+		Set<OpenDcsApiRoles> roles = BasicAuthCheck.getUserRoles(credentials.getUsername());
+		OpenDcsPrincipal principal = new OpenDcsPrincipal(credentials.getUsername(), roles);
+		HttpSession oldSession = request.getSession(false);
+		if(oldSession != null)
+		{
+			oldSession.invalidate();
+		}
+		HttpSession session = request.getSession(true);
+		session.setAttribute(OpenDcsPrincipal.USER_PRINCIPAL_SESSION_ATTRIBUTE, principal);
+		return ApiHttpUtil.createResponse("Authentication Successful.");
```

Comment:
I think we should be returning these messages in JSON format.  This will cause an issue with clients expecting content-type = application/json.
If we standardize on our responses being sent in JSON, then the apps using it can always expect the data to come back in that format.

I also realize that several of these endpoints (such as deletes) have been returning in plain text.

Reply:
I agree with that. Though in the case of deletes, I believe they shouldn't actually return a body, but the code should be I think 204 (whatever "No Content" is).

Reply:
Created an issue to track the deletes. For right now, I'll wrap all the `ApiHttpUtil.createResponse(String)` calls with the `Status` entity: https://github.com/opendcs/rest_api/issues/195

Reply:
@wjonassen -- can you please resolve this if your comment has been addressed?